helm-chart/todoapp/templates/hpa.yml

```diff
@@ -0,0 +1,29 @@
+{{if eq .Values.todoapp.hpa.enabled true}}
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{.Chart.Name}}
+  namespace: {{ .Values.todoapp.namespace }}
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: {{.Release.Name}}-{{.Chart.Name}}
+  {{- with .Values.todoapp.hpa}}
+  minReplicas: {{.minReplicas}}
+  maxReplicas: {{.maxReplicas}}
+  metrics:
+  - type: Resource
+    resource:
+      name: cpu
+      target:
+        type: Utilization
+        averageUtilization: {{.cpu.averageUtilization}}
```

Comment:
Ensure that `cpu.averageUtilization` is defined in your `values.yaml` file under `todoapp.hpa`. Missing values could lead to incorrect CPU utilization targets for autoscaling.